lib/shared/store.js

```diff
@@ -486,7 +487,46 @@ const storeReducer = (state = DEFAULT_STATE, action) => {
   }
 };
 
-module.exports = _.merge(redux.createStore(storeReducer, DEFAULT_STATE), {
-  Actions: ACTIONS,
-  Defaults: DEFAULT_STATE
-});
+// We're not in Electron
+if (typeof process === 'undefined') {
+  console.log('PLAIN REDUX');
+  module.exports = _.merge(redux.createStore(storeReducer, DEFAULT_STATE), {
+    Actions: ACTIONS,
+    Defaults: DEFAULT_STATE
+  });
+
+// We are in the renderer process of Electron
+} else if (_.get(process, [ 'type' ]) === 'renderer') {
+  console.log('RENDERER REDUX');
+  const store = _.merge(redux.createStore(storeReducer, DEFAULT_STATE, redux.applyMiddleware(
+    electronRedux.forwardToMain,
+    store => next => action => {
+      console.log('PREV', action);
+      const result = next(action);
+      console.log('NEXT', store.getState());
+      return result;
+    }
+  )), {
+    Actions: ACTIONS,
+    Defaults: DEFAULT_STATE
+  });
+
+  electronRedux.replayActionRenderer(store);
+
+  module.exports = store;
+
+// We are in the main Electron process
+} else if (process) {
+  console.log('MAIN REDUX');
+  const store = _.merge(redux.createStore(storeReducer, DEFAULT_STATE, redux.applyMiddleware(
+    electronRedux.triggerAlias,
+    electronRedux.forwardToRenderer
+  )), {
+    Actions: ACTIONS,
+    Defaults: DEFAULT_STATE
+  });
```

Comment:
Looks like this `{ Actions: ACTIONS, Defaults: DEFAULT_STATE }` is repeated 3 separate times. I dunno what this code is doing, but would it be worth making a new constant for that?